code/script.js

```diff
@@ -253,45 +283,100 @@ const setSecret = () => {
 const selectQuestion = () => {
   const category = questions.options[questions.selectedIndex].parentNode.label
   // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
+  const value = questions.options[questions.selectedIndex].value
 
   if (category === 'hair color') {
     currentQuestion = {
       attribute: 'hairColor',
-      // value: ,
-      // 👆 add the value from the input here
-      category: category,
+      value: value,
+      category: category
     }
   } else if (category === 'eye color') {
-    // Set this up your self
+    currentQuestion = {
+      attribute: "eyeColor",
+      value: value,
+      category: category,
+    }
+  } else if (category === 'character') {
+    currentQuestion = {
+      attribute: "character",
+      value: value,
+      category: category,
+    }
   } else if (category === 'accessories') {
     currentQuestion = {
-      //attribute: ,
+      attribute: value,      
       // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
       value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
       category: category,
     }
   } else if (category === 'other') {
-    // Set this up your self (should be same structure as above)
+    currentQuestion = {
+      attribute: value,
+      value: true,
+      category: category,
+    }
+  }
+}
+
+const countingGuesses = () => {
+  //numGuess += 1
+  infoText.innerHTML = `
+  number of guesses: ${numGuess}
+  `
+  if (numGuess === 3) {
+    alert(
+      "You have made 3 guesses, now you must choose the person"
+    )
+     
+  } else {
+        //Do nothing
   }
 }
 
 // This function should be invoked when you click on 'Find Out'.
 const checkQuestion = () => {
-  // Compare the currentQuestion with the secret person.
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
+  let keep
+  
+  if (currentQuestion.attribute === "hairColor" && currentQuestion.value === secret.hairColor) {
+    keep = true
+  } 
+  else if (currentQuestion.attribute === "eyeColor" && currentQuestion.value === secret.eyeColor) {
+    keep = true
+  }
+  else if (currentQuestion.attribute === "character" && currentQuestion.value === secret.character) {
+    keep = true
+  }  
+  else if (currentQuestion.attribute === "glasses" && currentQuestion.value === secret.glasses) {
+    keep = true
+  }
+  else if (currentQuestion.attribute === "hat" && currentQuestion.value === secret.hat) {
+    keep = true
+  }
+  else if (currentQuestion.attribute === "smoker" && currentQuestion.value === secret.smoker) {
+    keep = true
+  } else {
+    keep = false
+  }  
+
+  filterCharacters(keep)
 }
 
 // It'll filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
+  console.log(keep)
```

Comment:
line 375 maybe good to take away console.log here?

Reply:
Yes remember to always clean your code from all console.log()s and commented out code blocks 🤓